From fd459401fb557946349bf3a70a47a5a2fa6934a3 Mon Sep 17 00:00:00 2001
From: Aishwarya R <quic_aisr@quicinc.com>
Date: Tue, 1 Mar 2022 15:04:31 +0530
Subject: [PATCH] ath12k: Fixing typecast error for 32bit

Fix typecast error such as cast to pointer from integer
of different size and GENMASK error for 32bit

Signed-off-by: Aishwarya R <quic_aisr@quicinc.com>
---
drivers/net/wireless/ath/ath12k/dp.h     | 2 +-
drivers/net/wireless/ath/ath12k/dp_rx.c  | 4 ++--
drivers/net/wireless/ath/ath12k/dp_tx.c  | 2 +-
drivers/net/wireless/ath/ath12k/hal_rx.c | 2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/dp.h
+++ b/drivers/net/wireless/ath/ath12k/dp.h
@@ -1883,7 +1883,7 @@ enum vdev_stats_offload_timer_duration {
 #define HTT_H2T_VDEV_TXRX_HDR_INTERVAL		GENMASK(24, 17)
 #define HTT_H2T_VDEV_TXRX_HDR_RESET_STATS	BIT(31)
 #define HTT_H2T_VDEV_TXRX_LO_BITMASK		GENMASK(31, 0)
-#define HTT_H2T_VDEV_TXRX_HI_BITMASK		GENMASK(63, 32)
+#define HTT_H2T_VDEV_TXRX_HI_BITMASK		GENMASK_ULL(63, 32)
 
 struct htt_h2t_msg_type_vdev_txrx_stats_req {
 	u32 hdr;
